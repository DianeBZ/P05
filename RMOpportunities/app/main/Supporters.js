import React, { Component } from 'react';
import { Text, Image, StyleSheet, View,ScrollView, Dimensions, TouchableHighlight, BackAndroid} from 'react-native';

import Translation from './Translation';


var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;

export default class Company extends Component {
  render() {
    BackAndroid.addEventListener('Back',this.onBackAndroid);
    return (
      <ScrollView>
		<View style={styles.container}>
			<Text style={styles.title1}>{Translation[lang].confidence1}</Text>
			<Text style={styles.title2}>{Translation[lang].confidence2}</Text>
			<View style={styles.separation}></View>
			<View style={styles.containerH}>
				<View style = {styles.decoupage}>
					<Image source={require('../img/directchimie.png')} style = {styles.logoDC} />
					<Text style = {styles.texteCarreau}>Direct Chimie</Text>
				</View>
				<View style = {styles.decoupage}>
					<Image source={require('../img/basf.png')} style = {styles.logoBASF} />
					<Text style = {styles.texteCarreau}>BASF</Text>
				</View>
			</View>
			<View style = {styles.decoupage}>
				<Image source={require('../img/kimibiz.png')} style = {styles.logoKimibiz} />
				<Text style = {styles.texteCarreau}>Kimibiz</Text>
			</View>
			
			<Text style={styles.title1}>{Translation[lang].supporters1}</Text>
			<Text style={styles.title2}>{Translation[lang].supporters2}</Text>
			<View style={styles.separation}></View>
			<View style={styles.containerH}>
				<View style = {styles.decoupageS}>
					<Image source={require('../img/axelera.png')} style = {styles.logoAxe} />
					<Text style = {styles.titreCarreau}>Axelera</Text>
					<Text style = {styles.texteCarreau}>{Translation[lang].axelera}</Text>
				</View>
				<View style = {styles.decoupageS}>
					<Image source={require('../img/uic-idf.png')} style = {styles.logoUIC} />
					<Text style = {styles.titreCarreau}>UIC Ile-de-France </Text>
					<Text style = {styles.texteCarreau}>{Translation[lang].uic}</Text>
				</View>
			</View>
			<View style={styles.containerH}>
				<View style = {styles.decoupageS}>
					<Image source={require('../img/ufcc.png')} style = {styles.logoBASF} />
					<Text style = {styles.titreCarreau}>UFCC </Text>
					<Text style = {styles.texteCarreau}>{Translation[lang].ufcc}</Text>
				</View>
				<View style = {styles.decoupageS}>
					<Image source={require('../img/usein.png')} style = {styles.logoUse} />
					<Text style = {styles.titreCarreau}>Use'In </Text>
					<Text style = {styles.texteCarreau}>{Translation[lang].usein}</Text>
				</View>
			</View>
		</View>
     </ScrollView>
    );
  }
  onBackAndroid = () => {
		const { navigator } = this.props;
		if (navigator && navigator.getCurrentRoutes().length > 1) {
			BackAndroid.removeEventListener('Back', this.onBackAndroid);
			navigator.pop();
			return true;
		} else {
			return false;
		}
  };
}

const styles = StyleSheet.create({
	container: {
      marginTop:height*0.1,
      flex:1,
   },
   
   separation:{
      borderTopWidth: 2,
      marginTop:10,
      marginBottom:1,
      marginRight:20,
      marginLeft:20,
      borderTopColor:"#A4D04A",
   },
   title1:{
	   textAlign:'center',
	   fontSize:20,
   },
   
   title2:{
	   textAlign:'center',
	   fontSize:25,
   },
   containerH: {
    flexDirection : 'row',
    alignItems:'center',
    justifyContent: 'center',
	
  },
  decoupage: {
    //width: width*0.4,
    //height:height*0.17,
	justifyContent: 'center',
    alignItems:'center',
	margin:width*0.07,
  },
  
  logoBASF:{
    width: width*0.25,
    height: width*0.2,
  },
  logoDC:{
    width: width*0.3,
    height: width*0.25,
  },
  logoKimibiz:{
    width: width*0.30,
    height: width*0.10,
  },
  
  decoupageS: {
    width: width*0.5,
    height:height*0.3,
	justifyContent: 'center',
    alignItems:'center',
	marginTop:width*0.07,
  },
  logoAxe:{
    width: width*0.4,
    height: width*0.07,
  },
  logoUIC:{
    width: width*0.27,
    height: width*0.18,
  },
  logoUse:{
    width: width*0.39,
    height: width*0.15,
  },
  texteCarreau:{
	  margin:width*0.02,
	  textAlign:'center',
  },
  titreCarreau:{
	  marginTop:width*0.05,
	  textAlign:'center',
	  fontWeight: "bold",	  
  },
});