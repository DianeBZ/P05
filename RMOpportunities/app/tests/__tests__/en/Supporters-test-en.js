// __tests__/Supporters-test-en.js
import 'react-native';
import React from 'react';

global.lang = 'en';
import Supporters from '../../../main/Supporters';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <Supporters />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
