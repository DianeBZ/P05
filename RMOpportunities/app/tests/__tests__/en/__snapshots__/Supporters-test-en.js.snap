// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "height": 106.72,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#F2F2F2",
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        They trust us
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 25,
            "textAlign": "center",
          }
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "borderTopColor": "#A4D04A",
            "borderTopWidth": 2,
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 10,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 52.50000000000001,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 187.5,
                "width": 225,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "margin": 15,
                "textAlign": "center",
              }
            }
          >
            Direct Chimie
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 52.50000000000001,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 150,
                "width": 187.5,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "margin": 15,
                "textAlign": "center",
              }
            }
          >
            BASF
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "margin": 52.50000000000001,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 75,
              "width": 225,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "margin": 15,
              "textAlign": "center",
            }
          }
        >
          Kimibiz
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        They support us
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 25,
            "textAlign": "center",
          }
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "borderTopColor": "#A4D04A",
            "borderTopWidth": 2,
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 10,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 400.2,
              "justifyContent": "center",
              "marginTop": 52.50000000000001,
              "width": 375,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 52.50000000000001,
                  "width": 300,
                }
              }
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontWeight": "bold",
                "marginTop": 37.5,
                "textAlign": "center",
              }
            }
          >
            Axelera
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "margin": 15,
                "textAlign": "center",
              }
            }
          >
            Cluster for the chemical and environmental sectors
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 400.2,
              "justifyContent": "center",
              "marginTop": 52.50000000000001,
              "width": 375,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 135,
                  "width": 202.5,
                }
              }
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontWeight": "bold",
                "marginTop": 37.5,
                "textAlign": "center",
              }
            }
          >
            UIC Ile-de-France 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "margin": 15,
                "textAlign": "center",
              }
            }
          >
            French professional organization of chemistry in Ile-de-France
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 400.2,
              "justifyContent": "center",
              "marginTop": 52.50000000000001,
              "width": 375,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 150,
                  "width": 187.5,
                }
              }
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontWeight": "bold",
                "marginTop": 37.5,
                "textAlign": "center",
              }
            }
          >
            UFCC 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "margin": 15,
                "textAlign": "center",
              }
            }
          >
            French Union for Chemical Trade
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 400.2,
              "justifyContent": "center",
              "marginTop": 52.50000000000001,
              "width": 375,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 112.5,
                  "width": 292.5,
                }
              }
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontWeight": "bold",
                "marginTop": 37.5,
                "textAlign": "center",
              }
            }
          >
            Use'In 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "margin": 15,
                "textAlign": "center",
              }
            }
          >
            University Jean-Monnet Incubator - Saint-Etienne
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
